### PR TITLE
如果构造函数上有一个[OpCode],替换New Array操作

### PR DESCRIPTION
**Pull request by lightszero**
实现 new Map()操作

### CODE REVIEW

File: neon/MSIL/Conv_Multi.cs [hunk 1]
```diff
@@ -1011,6 +1011,29 @@ private int _ConvertNewObj(OpCode src, NeoMethod to)
...
+                {
+                    foreach (var attr in m.CustomAttributes)
+                    {
+                        if (attr.AttributeType.Name == "OpCodeAttribute")
+                        {
+                            //var _type = attr.ConstructorArguments[0].Type;
+                            var value = (byte)attr.ConstructorArguments[0].Value;
+                            VM.OpCode v = (VM.OpCode)value;
+
+                            _Insert1(VM.OpCode.NEWMAP, null, to);
```
Comment (erikzhang) [inline neon/MSIL/Conv_Multi.cs]: Is this a hardcoded `NEWMAP`?

Comment (lightszero) [reply neon/MSIL/Conv_Multi.cs]: err，this is a mistake

File: neon/MSIL/Conv_Multi.cs [hunk 2]
```diff
@@ -1011,6 +1011,26 @@ private int _ConvertNewObj(OpCode src, NeoMethod to)
...
             var type = _type.Resolve();
+
+            //如果构造函数上有一个[OpCode],替换New Array操作
+            foreach (var m in type.DeclaringType.Methods)
+            {
+                if (m.IsConstructor && m.HasCustomAttributes)
+                {
+                    foreach (var attr in m.CustomAttributes)
+                    {
+                        if (attr.AttributeType.Name == "OpCodeAttribute")
```
Comment (erikzhang) [inline neon/MSIL/Conv_Multi.cs]: Can we support other attributes such as `AppcallAttribute`?

Comment (lightszero) [reply neon/MSIL/Conv_Multi.cs]: yes

File: neon/MSIL/Conv_Multi.cs [hunk 3]
```diff
@@ -1019,7 +1019,9 @@ private int _ConvertNewObj(OpCode src, NeoMethod to)
                 {
                     foreach (var attr in m.CustomAttributes)
                     {
-                        if (attr.AttributeType.Name == "OpCodeAttribute")
+                        if (attr.AttributeType.Name == "OpCodeAttribute" ||
+                            attr.AttributeType.Name == "AppcallAttribute" ||
+                            attr.AttributeType.Name == "SyscallAttribute")
                         {
                             //var _type = attr.ConstructorArguments[0].Type;
                             var value = (byte)attr.ConstructorArguments[0].Value;
```
Comment (erikzhang) [inline neon/MSIL/Conv_Multi.cs]: I don't think this will work for `AppcallAttribute` or `SyscallAttribute`.

Comment (lightszero) [reply neon/MSIL/Conv_Multi.cs]: 。。。you said add that

**Pull request merged**